beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -49,81 +49,67 @@ func (s *Service) processPendingBlocksQueue() {
 	})
 }
 
-// processes the block tree inside the queue
+// processPendingBlocks validates, processes, and broadcasts pending blocks.
 func (s *Service) processPendingBlocks(ctx context.Context) error {
 	ctx, span := trace.StartSpan(ctx, "processPendingBlocks")
 	defer span.End()
 
-	pids := s.cfg.p2p.Peers().Connected()
+	// Validate pending slots before processing.
 	if err := s.validatePendingSlots(); err != nil {
 		return errors.Wrap(err, "could not validate pending slots")
 	}
-	ss := s.sortedPendingSlots()
-	var parentRoots [][32]byte
 
-	span.AddAttributes(
-		trace.Int64Attribute("numSlots", int64(len(ss))),
-		trace.Int64Attribute("numPeers", int64(len(pids))),
-	)
+	// Sort slots for ordered processing.
+	sortedSlots := s.sortedPendingSlots()
+
+	span.AddAttributes(trace.Int64Attribute("numSlots", int64(len(sortedSlots))), trace.Int64Attribute("numPeers", int64(len(s.cfg.p2p.Peers().Connected()))))
 
 	randGen := rand.NewGenerator()
-	for _, slot := range ss {
-		// process the blocks during their respective slot.
-		// otherwise wait for the right slot to process the block.
+	var parentRoots [][32]byte
+
+	// Iterate through sorted slots.
+	for _, slot := range sortedSlots {
+		// Skip processing if current slot is ahead.
```

Comment:
if sorted slot is ahead of current slot ( i assume you meant that the current slot processed here is more than the current slot stored in cfg ) just weird wording, not sure about the best language to improve.